src/pages/Swap/index.tsx

```diff
@@ -296,6 +298,20 @@ export default function Swap() {
   ])
 
   const [showAddRecipient, setShowAddRecipient] = useState<boolean>(false)
+  const [isMaxAmountInput, setIsMaxAmountInput] = useState<boolean>(false)
+  const [isMaxAmountOutput, setIsMaxAmountOutput] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (isMaxAmountInput && maxAmountInput) {
+      setIsMaxAmountInput(false)
+      onUserInput(Field.INPUT, maxAmountInput.toExact())
+    }
+
+    if (isMaxAmountOutput && maxAmountOutput) {
+      setIsMaxAmountOutput(false)
+      onUserInput(Field.OUTPUT, maxAmountOutput.toExact())
+    }
+  }, [isMaxAmountInput, isMaxAmountOutput, maxAmountInput, maxAmountOutput, onUserInput])
```

Comment:
Could logic for this be contained inside the currency input panel? So its easier to extend to all input panels since if we want to add this to liquidity down the line we would have to repeat this code there as well.
- Like maybe having that logic here if possible?
![Screenshot 2022-05-27 at 19 08 13](https://user-images.githubusercontent.com/33226956/170750808-85569945-e71f-4d0a-b216-9e7f1fd7a301.png)

Reply:
Thanks for your suggestion @Mi-Lan, it's better now! Fixed in [e0b2cc0](https://github.com/levelkdev/swapr-dapp/pull/1013/commits/e0b2cc0ab58fe8886754c9d4d66ad933f11f4183)